### ISSUE TITLE
Why import type syntax marked as invalid?

### ISSUE THREAD

**Issue by krutoo** [gajus/eslint-plugin-jsdoc#677]
Hi, i have two files with code:
```js
// event.js
export class CustomEvent {}

// index.js
/**
 * @param {import('./event').CustomEvent} event Event.
 */
export function main (event) {
  console.log(event);
}
```

and config like:
```js
module.exports = {
  // ...
  rules: {
    // ...
    'jsdoc/check-param-names': 'error',
    'jsdoc/valid-types': 'error',
  },
};
```

in terminal i have error:
```bash
> eslint ./src

/home/petrov_dm/Projects/eslint-plugin-jsdoc-tests/src/index.js
  2:0  error  Syntax error in type: import('./event').CustomEvent  jsdoc/valid-types

✖ 1 problem (1 error, 0 warnings)
```

**there is repo with clean  example of code:**
https://github.com/krutoo/eslint-plugin-jsdoc-tests

Why this code is marked as with syntax error? 
(this worked differently in previous plugin versions)

`readme.md` tells that is valid type syntax:
https://github.com/gajus/eslint-plugin-jsdoc#why-not-capital-case-everything

> The following patterns are not considered problems:

with this code example:
```js
/**
 * @param {import('./foo').bar.baz} foo
 */
function qux(foo) {
}
```

**Comment by brettz9** [comment]
You need either `parser: '@typescript-eslint/parser'` or `settings: {jsdoc: {mode: 'typescript'}}` in your ESLint config since plain jsdoc does not support that import syntax; see, e.g., https://github.com/jsdoc/jsdoc/issues/1632 and https://github.com/jsdoc/jsdoc/issues/1645 . The reason the README accepts that without settings is because for the `check-types` rule, it is only checking for things like the casing of "Object".

Or as the docs for `check-types` put it:

> Note that if there is an error [parsing](https://github.com/jsdoctypeparser/jsdoctypeparser)
types for a tag, the function will silently ignore that tag, leaving it to
the `valid-types` rule to report parsing errors.

For `valid-types`, you need proper parsing suitable for the `mode` (which defaults to "jsdoc" which supports `module:` syntax instead).
